src/go/rpk/pkg/cli/cluster/partitions/list.go

```diff
@@ -14,194 +21,107 @@ import (
 	"github.com/twmb/types"
 )
 
-func newListPartitionMovementsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+func newListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 	var (
-		completion       int
-		all              bool
-		human            bool
-		partitions       []string
-		response         []adminapi.ReconfigurationsResponse
-		filteredResponse []adminapi.ReconfigurationsResponse
+		disabledOnly  bool
+		withInternals bool
 	)
 	cmd := &cobra.Command{
-		Use:   "move-status",
-		Short: "Show ongoing partition movements",
-		Long:  helpListMovement,
+		Use:     "list [TOPICS]",
+		Aliases: []string{"ls"},
+		Short:   "List partitions in the cluster",
+		Long: `List partitions in the cluster
+
+This commands lists the cluster-level metadata of all partitions in the cluster, 
+you may filter the topics you want to list by passing them as argument to this
+command.
+
+The REPLICAS column displayed in the output table contains a list of replicas
+assignments in the form of: <Node-ID>:<Core>.
+
+EXAMPLES
+
+List all partitions in the cluster.
+  rpk cluster partitions list
+
+List all partitions in the cluster, filtering for topic foo and bar.
+  rpk cluster partitions list foo bar
+
+List only disabled partitions.
+  rpk cluster partitions list --disabled
+
+List all in json format, but include also internal topics.
+  rpk cluster partition list --format json --with-internals
+`,
 		Run: func(cmd *cobra.Command, topics []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]adminapi.ClusterPartition{}); ok {
+				out.Exit(h)
+			}
 			p, err := p.LoadVirtualProfile(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
 			out.CheckExitCloudAdmin(p)
 
-			// If partition(s) is specified but no topic(s) is specified, exit.
-			if len(topics) <= 0 && len(partitions) > 0 {
-				out.Die("specify at least one topic when --partition is used, exiting.")
-			}
-
 			cl, err := adminapi.NewClient(fs, p)
 			out.MaybeDie(err, "unable to initialize admin client: %v", err)
 
-			response, err = cl.Reconfigurations(cmd.Context())
-			out.MaybeDie(err, "unable to list partition movements: %v\n", err)
-
-			if len(response) == 0 {
-				out.Exit("There are no ongoing partition movements.")
-			}
-
-			for _, t := range topics {
-				nt := strings.Split(t, "/")
-				if len(nt) > 2 {
-					fmt.Printf("invalid format for topic %s, skipping.\n", t)
-					continue
+			var partitions []adminapi.ClusterPartition
+			if len(topics) == 0 {
+				partitions, err = cl.AllClusterPartitions(cmd.Context(), withInternals, disabledOnly)
+				out.MaybeDie(err, "unable to query all partitions in the cluster: %v", err)
+			} else {
+				if withInternals {
+					out.Die("--with-internals not available when listing single topics.\n%v", cmd.UsageString())
```

Comment:
I'd use the same flag name we use in `rpk topic list`, `-i`, `--internal`

Reply:
Ok, but then should we rename `disabled` to `disabled-only` ? `with-internals`  add the internal topics in addition to the rest of the topics. But `disabled` only shows the disabled partitions.

Reply:
Actually though I'm not sure --internal is a great flag, I'd maybe rename this to --with-internal (drop the s), rename the `rpk topic list` flag, and hide the old `rpk topic list` flag. I wonder what Confluent tools name this flag.

Reply:
In kafka_topics.sh:

```
--exclude-internal                       exclude internal topics when running
                                          list or describe command. The
                                          internal topics will be listed by
                                          default
```

Reply:
I'm struggling to think of a scenario where we wouldn't want internal partitions to be displayed -- this is an admin command after all, in service of admins figuring out which partitions are disabled.

Reply:
This was removed and enabled by default :+1: